ocaml/nbd/src/main.ml

```diff
@@ -177,14 +177,10 @@ let main port certfile _ =
   in
   (* Log unexpected exceptions *)
   let () =
-    Lwt_main.run
-      (Lwt.catch t (fun e ->
-           Lwt_log.fatal_f "Caught unexpected exception: %s"
-             (Printexc.to_string e)
-           >>= fun () -> Lwt.fail e
-       )
-      )
+    try Lwt_main.run (ignore_exn_delayed t ())
+    with e -> Lwt_log.ign_fatal (Printexc.to_string e)
```

Comment:
I think this changes the exitcode of the process which isn't necessarily right. If we had an exception we want a non-zero exitcode, like before.

Reply:
Try adding a call to 'exit 3' here or something similar after logging the exception. Although the Lwt main loop is not running anymore at this point so I doubt we'll succeed in logging it.
Maybe it'd be better to use Lwt_main.run (Lwt_log.fatal...)